contracts/clients/src/BlsSigner.ts

```diff
@@ -1,14 +1,77 @@
 /* eslint-disable camelcase */
-import { ethers, BigNumber, Signer, Bytes } from "ethers";
-import { Deferrable, hexlify, isBytes, RLP } from "ethers/lib/utils";
-import { AggregatorUtilities__factory } from "../typechain-types";
+import { ethers, BigNumber, Signer, Bytes, BigNumberish } from "ethers";
+import {
+  AccessListish,
+  BytesLike,
+  Deferrable,
+  hexlify,
+  isBytes,
+  RLP,
+} from "ethers/lib/utils";
 
 import BlsProvider from "./BlsProvider";
 import BlsWalletWrapper from "./BlsWalletWrapper";
+import addSafetyPremiumToFee from "./helpers/addSafetyDivisorToFee";
 import { ActionData, bundleToDto } from "./signer";
 
 export const _constructorGuard = {};
 
+/**
+ * @param to - 20 Bytes The address the transaction is directed to. Undefined for creation transactions
+ * @param value - the value sent with this transaction
+ * @param gas - transaction gas limit
+ * @param maxPriorityFeePerGas - miner tip aka priority fee
+ * @param maxFeePerGas - the maximum total fee per gas the sender is willing to pay (includes the network/base fee and miner/priority fee) in wei
+ * @param data - the hash of the invoked method signature and encoded parameters
+ * @param nonce - integer of a nonce. This allows overwriting your own pending transactions that use the same nonce
+ * @param chainId - chain ID that this transaction is valid on
+ * @param accessList - EIP-2930 access list
+ */
+export type Transaction = {
+  to?: string;
+  value?: BigNumberish;
+  gas?: BigNumberish;
+  maxPriorityFeePerGas?: BigNumberish;
+  maxFeePerGas?: BigNumberish;
+  data?: BytesLike;
+  nonce?: BigNumberish;
+  chainId?: number;
+  accessList?: Array<AccessListish>;
+};
+
+/**
+ * @param gas - transaction gas limit
+ * @param maxPriorityFeePerGas - miner tip aka priority fee
+ * @param maxFeePerGas - the maximum total fee per gas the sender is willing to pay (includes the network/base fee and miner/priority fee) in wei
+ * @param nonce - integer of a nonce. This allows overwriting your own pending transactions that use the same nonce
+ * @param chainId - chain ID that this transaction is valid on
+ * @param accessList - EIP-2930 access list
+ */
+export type BatchOptions = {
+  gas?: BigNumberish;
+  maxPriorityFeePerGas: BigNumberish;
+  maxFeePerGas: BigNumberish;
+  nonce: BigNumberish;
+  chainId: number;
+  accessList?: Array<AccessListish>;
+};
+
+/**
+ * @param transactions - an array of transaction objects
+ * @param batchOptions - optional batch options taken into account by SC wallets
+ */
+export type TransactionBatch = {
+  transactions: Array<Transaction>;
+  batchOptions?: BatchOptions;
```

Comment:
Not really utilising this optional property. Tbh slightly confused by what it's purpose is from the [draft standard](https://gist.github.com/mikhailxyz/773ba5196b5a7046423592204e5bc5c4). My interoperation is that these values help smart contract wallets deal with figuring out batch transaction options from multiple transactions, but not sure if that's correct

Reply:
I think using this spec we need to accept them, but some of them we will be unable to pass forward to the agg. `gas` can be used with latest contract change.

Reply:
Done. Added a helper function (_validateBatchOptions) that validates the batch options if they are passed as an argument. It checks the chain id is correct and ensures the nonce is the correct type.